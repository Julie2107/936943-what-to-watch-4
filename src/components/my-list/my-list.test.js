import React from "react";
import renderer from "react-test-renderer";
import configureStore from 'redux-mock-store';
import NameSpace from "../../reducer/name-space";
import MyList from "./my-list";
import {Provider} from "react-redux";
import { Router } from "react-router-dom";
import history from "../../history";

const mockStore = configureStore([]);

const mocks = [
  {title: `Mindhunter`, poster: `http://placekitten.com/245/175`, cover: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseYear: 2000, src: ``, rating: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, isFavourite: true},
  {title: `Aviator`, poster: `http://placekitten.com/245/175`, cover: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseYear: 2000, src: ``, rating: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, isFavourite: true},
  {title: `Pulp Fiction`, poster: `http://placekitten.com/245/175`, cover: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseYear: 2000, src: ``, rating: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, isFavourite: true},
  {title: `Midnight Special`, poster: `http://placekitten.com/245/175`, cover: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseYear: 2000, src: ``, rating: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, isFavourite: true},
];


it(`Render myList`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      myListMovies: mocks,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      isAuthError: false
    }
  });

  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <MyList
              movies={store.movies}
              onTitleClick={() => {}}
              authorizationStatus={`AUTH`}
              isAuthError={false}
            />
          </Provider>
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
