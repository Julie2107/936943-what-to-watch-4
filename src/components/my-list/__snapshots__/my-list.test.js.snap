// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render myList 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="/img/avatar.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            className="player__video"
            muted={true}
            poster="http://placekitten.com/245/175"
            src=""
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/undefined"
            onClick={[Function]}
          >
            Mindhunter
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            className="player__video"
            muted={true}
            poster="http://placekitten.com/245/175"
            src=""
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/undefined"
            onClick={[Function]}
          >
            Aviator
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            className="player__video"
            muted={true}
            poster="http://placekitten.com/245/175"
            src=""
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/undefined"
            onClick={[Function]}
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            className="player__video"
            muted={true}
            poster="http://placekitten.com/245/175"
            src=""
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/undefined"
            onClick={[Function]}
          >
            Midnight Special
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="main.html"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
